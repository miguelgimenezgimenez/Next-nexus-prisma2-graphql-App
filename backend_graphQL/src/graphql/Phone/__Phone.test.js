const { createTestContext } = require('../../../tests/__helpers')
const ctx = createTestContext()
it('ensures that a phone can be created and updated', async () => {
  // Create a new draft
  const draftResult = await ctx.client.request(`
    mutation {
      addPhone(name: "Nexus", brand_id: 1) {  
        name
        id
      }
    }
  `)


  expect(draftResult).toMatchSnapshot(`
    Object {
      "addPhone": Object {
        "id":${draftResult.addPhone.id},
        "name": "Nexus"
      },
    }
  `)
  const publishResult = await ctx.client.request(
    `
    mutation updatePhone($id: Int!,$brand_id: Int!,$name:String!) {
      updatePhone(id: $id, name:$name, brand_id:$brand_id) {
        id
        name
      }
    }
  `,
    { id: draftResult.addPhone.id, brand_id: 1, name: 'UpdatedName' },
  )

  expect(publishResult).toMatchSnapshot(`
    Object {
      "updatePhone": Object {
        "id": ${draftResult.addPhone.id},
        "name": "UpdatedName"
      }
    }
  `)
  const persistedData = await ctx.db.phone.findMany()

  expect(persistedData).toMatchSnapshot(`
     [
       {
        "brand_id": 1,
        "dimensions": null,
        "id": 1,
        "image": "https://fdn2.gsmarena.com/vv/bigpic/apple-iphone-12-pro--.jpg",
        "link": null,
        "name": "UpdatedName",
        "os": null,
        "storage": null
      }
    ]
  `)
})
