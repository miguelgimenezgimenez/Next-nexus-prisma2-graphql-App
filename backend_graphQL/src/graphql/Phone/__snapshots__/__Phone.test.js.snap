// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ensures that a phone can be created: 
     [
       {
        "brand_id": 1,
        "dimensions": null,
        "id": 1,
        "image": "https://fdn2.gsmarena.com/vv/bigpic/apple-iphone-12-pro--.jpg",
        "link": null,
        "name": "UpdatedName",
        "os": null,
        "storage": null
      }
    ]
   1`] = `
Array [
  Object {
    "brand_id": 1,
    "dimensions": null,
    "id": 1,
    "image": "https://fdn2.gsmarena.com/vv/bigpic/apple-iphone-12-pro--.jpg",
    "link": null,
    "name": "UpdatedName",
    "os": null,
    "storage": null,
  },
]
`;

exports[`ensures that a phone can be created: 
    Array [
      Object {
        "brand_id": 1,
        "dimensions": null,
        "id": 1,
        "image": "https://fdn2.gsmarena.com/vv/bigpic/apple-iphone-12-pro--.jpg",
        "link": null,
        "name": "UpdatedName",
        "os": null,
        "storage": null
      }
    ]
   1`] = `
Array [
  Object {
    "brand_id": 1,
    "dimensions": null,
    "id": 1,
    "image": "https://fdn2.gsmarena.com/vv/bigpic/apple-iphone-12-pro--.jpg",
    "link": null,
    "name": "UpdatedName",
    "os": null,
    "storage": null,
  },
]
`;

exports[`ensures that a phone can be created: 
    Object {
      "addPhone": Object {
        "id":1,
        "name": "Nexus"
      },
    }
   1`] = `
Object {
  "addPhone": Object {
    "id": 1,
    "name": "Nexus",
  },
}
`;

exports[`ensures that a phone can be created: 
    Object {
      "updatePhone": Object {
        "id": 1,
        "name": "UpdatedName"
      }
    }
   1`] = `
Object {
  "updatePhone": Object {
    "id": 1,
    "name": "UpdatedName",
  },
}
`;
